File: Sorting/Java/BubbleSort.java
import java.util.Arrays;

// We keep swapping to keep max element at its right index

public class BubbleSort {
    public static void main(String[] args) {
        int arr[] = {-1,9,-2,3,4,12,0};
        bubbleSort(arr);
        System.out.println(Arrays.toString(arr));
    }
    static void bubbleSort(int[] arr){
        boolean swapped;
        // run the steps n - 1 times
        for (int i = 0; i < arr.length - 1; i++) {
            swapped = false;
            // for each step. max item will come at the last respective index
            for (int j = 1; j <= arr.length - i -1 ; j++) {
                // swap if item is smaller than previous item
                if (arr[j] < arr[j-1]){
                    //swap
                    int temp = arr[j];
                    arr[j] = arr[j-1];
                    arr[j-1] = temp;
                    swapped = true;
                }
            }
            // if there is no swap for i = 0, then array is sorted
            if(! swapped) {
                break;
            }
        }
    }
}

File: Sorting/JavaScript/BubbleSort.js
function bubbleSort(arr)
{
    let temp =0;
    //Take a temp variable assing it to 0
    for(let i=0;i<arr.length;i++)
    {
        //taking i loop for comparision
        for(let j=0;j<arr.length;j++)
        //taking j loop for comparision
        {
            if(arr[i] > arr[i+j])
            //if(54 is greater than 32 ) in first itreation 
            {
                temp = arr[i];
                // 0 is now 54
                arr[i] = arr[i +j];
                //54 is now 32
                arr[i + j ] =  temp;
                //32 is now 54
            }
        }
    }
    return arr;
    //sorted array returned
}
const arr = [54,32,34,76,23,12,11,56,88,43];

console.log(bubbleSort(arr));